utils/run_helpers.py

```diff
@@ -119,7 +119,7 @@ def add_experiment_config_to_blocks_(
 ) -> None:
     for model in flatten(pipeline.children()):
         model.run_context = RunContext(
-            project_name=copy(experiment.project_name),
-            run_name=copy(experiment.run_name),
```

Comment:
you don't need to copy strings, they're not passed as references - they're always copied when there's a reassignment. so they don't share an underlying object.

Reply:
https://www.delftstack.com/howto/python/python-copy-string/#:~:text=In%20Python%2C%20strings%20are%20immutable,copy%20of%20the%20original%20string.